pkg/crypto/crypto_test.go

```diff
@@ -252,3 +252,79 @@ func TestNewEthereumAddress(t *testing.T) {
 		t.Fatalf("address mismatch %x %x", address, expectAddress)
 	}
 }
+
+func TestNewOverlayFromEthereumAddress(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		ethAddress    string
+		overlay       uint64
+		hash          string
+		expectedHex   string
```

Comment:
Consider renaming it to `wantAddress` and using the `swarm.Address` instead of the string type. Then the fields can be initialized by using `swarm.MustParseHexAddress(...)` and compared with `wantAddress.Equal(gotAddress)`.